tests/generators/test_litellm.py

```diff
@@ -43,3 +44,11 @@ def test_litellm_openrouter():
     for item in output:
         assert isinstance(item, str)
     print("test passed!")
+
+
+def test_litellm_model_non_existence():
+    model_name = "non-existent-model"
+    generator = LiteLLMGenerator(name=model_name)
+    with pytest.raises(BadGeneratorException):
+        output = generator.generate("This should raise an exception")
+    assert "Exceptions on model non-existence raised by litellm should be bubbled up"
```

Comment:
Since there is no actual `litellm` instance reached by this test, I would expect there needs to be more configuration values passed to call the completion.

This test would need to be passed `api_base` and also a model name that can trigger `openai` detection:
```suggestion
def test_litellm_model_detection():
    custom_config = {
        "generators": {
            "litellm": {
                "api_base": "https://garak.example.com/v1",
            }
        }
    }
    model_name = "non-existent-model"
    generator = LiteLLMGenerator(name=model_name, config_root=custom_config)
    with pytest.raises(BadGeneratorException):
        generator.generate("This should raise an exception")
    generator = LiteLLMGenerator(name="openai/invalid-model", config_root=custom_config)
    with pytest.raises(BadGeneratorException):
        generator.generate("This should raise an exception")
```

The assert at end of the test is redundant although capturing the exception and asserting some unique expectation on it would make sense.

Reply:
This new test fails if we don't catch BadRequestError
litellm raises BadRequestError if we give a non existent model without a provider

```
E               litellm.exceptions.BadRequestError: litellm.BadRequestError: LLM Provider NOT provided. Pass in the LLM provider you are trying to call. You passed model=non-existent-model
E                Pass model as E.g. For 'Huggingface' inference endpoints pass in `completion(model='huggingface/starcoder',..)` Learn more: https://docs.litellm.ai/docs/providers

.venv/lib/python3.12/site-packages/litellm/utils.py:4827: BadRequestError
============================================================================================ short test summary info =============================================================================================
FAILED tests/generators/test_litellm.py::test_litellm_model_detection - litellm.exceptions.BadRequestError: litellm.BadRequestError: LLM Provider NOT provided. Pass in the LLM provider you are trying to call. You passed model=non-existent-model
```

Reply:
In my local tests an model without a provider detected would raise, `APIConnectionError`. The upstream library has changed behavior: 

```
litellm.exceptions.APIConnectionError: LLM Provider NOT provided. Pass in the LLM provider you are trying to call. You passed model=non-existent-model
 Pass model as E.g. For 'Huggingface' inference endpoints pass in `completion(model='huggingface/starcoder',..)` Learn more: https://docs.litellm.ai/docs/providers
```

As of `1.41.21` the raised error type changed to BadRequestError, please update `requirements.txt` and `pyproject.toml` to required `litellm>=1.41.21` and adjust the `try` block to catch (BadRequestError, AuthenticationError) to match the newer library package expectations.

Reply:
Thanks for this investigation @jmartin-tech
Updated the litellm version and adjusted the try block